src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSObject.References.cs

```diff
@@ -56,6 +64,30 @@ internal void ReleaseInFlight()
         }
 #endif
 
+#if FEATURE_WASM_THREADS
+        internal static void AssertThreadAffinity(object value)
+        {
+            if (value == null)
+            {
+                return;
+            }
+            if (value is JSObject jsObject)
+            {
+                if (jsObject.OwnerThreadId != Thread.CurrentThread.ManagedThreadId)
+                {
+                    throw new InvalidOperationException("The JavaScript object can be used only on the thread where it was created.");
+                }
+            }
+            if (value is JSException jsException)
+            {
+                if(jsException.jsException!=null && jsException.jsException.OwnerThreadId != Thread.CurrentThread.ManagedThreadId)
+                {
+                    throw new InvalidOperationException("The JavaScript object can be used only on the thread where it was created.");
```

Comment:
We don't want to surface the exception that the JsException contains? Why the message is 100% the same as for JsObject?
```suggestion
                    throw new InvalidOperationException("The JavaScript exception can be used only on the thread where it was created.");
```